specification/sections/02-providers.md

```diff
@@ -150,3 +151,85 @@ class MyProvider implements Provider {
 #### Requirement 2.3.3
 
 > In cases of abnormal execution, the `resolution details` structure's `error message` field **MAY** contain a string containing additional detail about the nature of the error.
+
+### 2.4 Initialization
+
+[![experimental](https://img.shields.io/static/v1?label=Status&message=experimental&color=orange)](https://github.com/open-feature/spec/tree/main/specification#experimental)
+
+#### Requirement 2.4.1
+
+> The `provider` interface **MUST** define an optional `initialize` function, which accepts the global `evaluation context` as an argument and performs initialization logic relevant to the provider.
```

Comment:
What's the benefit of forcing this initialize function, rather than doing the initialization implicitly in the constructor?

Reply:
The constructor can't have the evaluation context. At initialization, the SDK can provide a context to the provider if one has been set. It makes things a bit more consistent with respect to the `onContextSet` handler: the provider author now has a few well-defined points at which to perform bulk evaluation/caching if that's relevant to them.